src/lib/components/Swap/Output.tsx

```diff
@@ -33,32 +36,35 @@ interface OutputProps {
 }
 
 export default function Output({ disabled, children }: OutputProps) {
-  const input = useAtomValue(inputAtom)
-  const output = useAtomValue(outputAtom)
-  const setValue = useUpdateInputValue(outputAtom)
-  const setToken = useUpdateInputToken(outputAtom)
-  const balance = 123.45
+  const {
+    currencyBalances: { [Field.OUTPUT]: balance },
+    currencyAmounts: { [Field.INPUT]: inputCurrencyAmount, [Field.OUTPUT]: outputCurrencyAmount },
+  } = useSwapInfo()
+  const inputUSDC = useUSDCValue(inputCurrencyAmount)
+  const outputUSDC = useUSDCValue(outputCurrencyAmount)
+
+  const [typedOutputAmount, updateTypedOutputAmount] = useOutputAmount()
+  const [outputCurrency, updateOutputCurrency] = useOutputCurrency()
 
   const overrideColor = useAtomValue(colorAtom)
-  const dynamicColor = useCurrencyColor(output.token)
-  usePrefetchCurrencyColor(input.token) // extract eagerly in case of reversal
+  const dynamicColor = useCurrencyColor(undefined)
+  usePrefetchCurrencyColor(undefined) // extract eagerly in case of reversal
   const color = overrideColor || dynamicColor
-  const hasColor = output.token ? Boolean(color) || null : false
+  const hasColor = outputCurrency ? Boolean(color) || null : false
```

Comment:
Add a comment over this that the different states (true/null/false) allow smooth color transitions.